pkgs/servers/sql/pgcat/default.nix

```
@@ -3,46 +3,42 @@
 , fetchFromGitHub
 , rustPlatform
 , darwin
+, nix-update-script
+, testers
+, pgcat
 }:
 
 rustPlatform.buildRustPackage rec {
   pname = "pgcat";
-  version = "1.1.1";
+  version = "1.2.0";
 
   src = fetchFromGitHub {
     owner = "postgresml";
     repo = "pgcat";
     rev = "v${version}";
-    hash = "sha256-BERxdGgPk8POnhLsyy4lKV4LCoHsJTmv2OhAOz6CKKc=";
+    hash = "sha256-DHXUhAAOmPSt4aVp93I1y69of+MEboXJBZH50mzQTm8=";
   };
 
-  cargoHash = "sha256-GwcqR8pEvz42NEmcuXpcoPdChzRBYsDEnllX62T8ulQ=";
+  cargoHash = "sha256-QqwUEbWKSUuxzYjWpVpQuKZDiNib1gM2eZJ4y7XIzXY=";
 
   buildInputs = lib.optionals stdenv.isDarwin [
     darwin.apple_sdk.frameworks.Security
   ];
 
   checkFlags = [
     # requires network access
-    "--skip=dns_cache::CachedResolver::lookup_ip"
-    "--skip=dns_cache::CachedResolver::new"
-    "--skip=dns_cache::CachedResolver"
-    "--skip=dns_cache::tests::has_changed"
-    "--skip=dns_cache::tests::incorrect_address"
-    "--skip=dns_cache::tests::lookup_ip"
-    "--skip=dns_cache::tests::new"
-    "--skip=dns_cache::tests::thread"
-    "--skip=dns_cache::tests::unknown_host"
+    "--skip=dns_cache"
   ];
 
-  doInstallCheck = true;
-  installCheckPhase = ''
-    $out/bin/pgcat --version | grep "pgcat ${version}"
-  '';
+  passthru = {
+    updateScript = nix-update-script { };
+    tests.version = testers.testVersion { package = pgcat; };
+  };
 
   meta = with lib; {
     homepage = "https://github.com/postgresml/pgcat";
     description = "PostgreSQL pooler with sharding, load balancing and failover support";
+    changelog = "https://github.com/postgresml/pgcat/releases/tag/v${version}";
```

```suggestion
    changelog = "https://github.com/postgresml/pgcat/releases";
```

https://github.com/NixOS/nixpkgs/pull/336200 not withstanding, I **really** dislike using `<github_url>/releases/tag/<tag>` over `<github_url>/releases`. I get this is up to personal preference, but you can read more about this here: https://github.com/NixOS/nixpkgs/issues/336454